### ISSUE TITLE
Type error: Type 'i18n' is not assignable to type 'null'.

### ISSUE THREAD

**Issue by cubi76041** [isaachinman/next-i18next#1026]
## Describe the bug
**yarn build** is throwing **Type error: Type 'i18n' is not assignable to type 'null'.**

## Occurs in next-i18next version
8.0.6

## Steps to reproduce
Contents of **tsconfig.json**

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "baseUrl": ".",
    "noImplicitAny": false,
  },
  "exclude": [
    "node_modules",
    "dist",
    ".next",
    "out",
    "next.config.js",
    ".babelrc",
    "bundles",
    "coverage"
  ],
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/jest.setup.ts"]
}
```

Contents of **next-i18next.config.js**:
```ts
module.exports = {
  i18n: {
    locales: ['en', 'fr'],
    defaultLocale: 'en',
  },
  debug: true,
  localePath: path.resolve("./public/static/locales"),
};
```
Contents of **next.config.js**:

```js
const { i18n } = require("./next-i18next.config");

const nextConfig = {
  productionBrowserSourceMaps: true,
  analyzeServer: ["server", "both"].includes(process.env.BUNDLE_ANALYZE),
  analyzeBrowser: ["browser", "both"].includes(process.env.BUNDLE_ANALYZE),
  bundleAnalyzerConfig: {
    server: {
      analyzerMode: "static",
      reportFilename: "./bundles/server.html",
    },
    browser: {
      analyzerMode: "static",
      reportFilename: "./bundles/client.html",
    },
  },
  i18n,
  async rewrites() {
    return [
      {
        source: "/",
        destination: "/home",
      },
    ];
  },
};
```

Contents of **pages/_app.tsx**:

```ts
class WebApp extends App<AppWithStore> {
  render() {
    const { Component, pageProps, store } = this.props;
    return (
      <Provider store={store}>
        <Head>
          <title>{pageProps?.title || DEFAULT_TITLE}</title>
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1, viewport-fit=cover"
          />
        </Head>
        <Layout>
          <Component {...pageProps} />
        </Layout>
      </Provider>
    );
  }
}

export default withRedux(makeStore)(appWithTranslation(WebApp));
```

Contens of pages/home/index.tsx:

```ts
const HomePage: NextPage<IHomePage.IProps, IHomePage.InitialProps> = () => (
  <HomeTemplate />
);

export const getStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale, ["common"])),
    },
  };
};

export default HomePage;
```

## Expected behaviour
**yarn build** should build the Next.js project without throwing any errors.

## Screenshots
<img width="634" alt="Screen Shot 2021-03-05 at 4 32 27 PM" src="https://user-images.githubusercontent.com/4811998/110096526-9025d480-7dd0-11eb-97ea-c871e01a7e1b.png">

## OS (please complete the following information)
 - Device: Macbook Pro 2017, OSX 10.15.7
 - Node: v.14.15.3

**Comment by isaachinman** [comment]
Please provide a reproducible repository. Thanks!

**Comment by stevejhiggs** [comment]
I also had this issue. It appears to be caused by an overly broad include in tsconfig. In my case it was set to:

```
 "exclude": ["node_modules", ".next"],
  "include": ["**/*.(ts|tsx)"]
```

You would think this would include all ts files but nothing in the node_modules folder. However it turns out in this case typescript will still attempt to typecheck any .ts files in node_modules. Depending on your strictness settings in your tsconfig.json this may or may not cause issues. In my case it did.

One workaround is to make the include more specific but this would require people to alter there projects because of this package. A better fix would be to ship the types solely as .d.ts files and stop shipping /src.

This can be done pretty easily by using the "declaration" and "emitDeclarationOnly" properties of ts.config and I'm just working on a pr to do exactly that.

**Comment by cubi76041** [comment]
Yes, it was fixed in v8.1.0. Many thanks @isaachinman @stevejhiggs !!
